core/templates/dev/head/pages/suggestion_editor/ShowSuggestionModalForEditorViewService.js

```diff
@@ -64,38 +67,69 @@ oppia.factory('ShowSuggestionModalForEditorViewService', [
         },
         controller: 'ShowSuggestionModalForEditorView'
       }).result.then(function(result) {
-        ThreadDataService.resolveSuggestion(
-          activeThread.threadId, result.action, result.commitMessage,
-          result.reviewMessage, result.audioUpdateRequired, function() {
-            ThreadDataService.fetchThreads(function() {
-              setActiveThread(activeThread.threadId);
-            });
-            // Immediately update editor to reflect accepted suggestion.
-            if (
-              result.action === SuggestionModalService.ACTION_ACCEPT_SUGGESTION
-            ) {
-              var suggestion = activeThread.getSuggestion();
+        var suggestion = activeThread.getSuggestion();
+        var stateName = suggestion.stateName;
 
-              var stateName = suggestion.stateName;
-              var stateDict = ExplorationDataService.data.states[stateName];
-              var state = StateObjectFactory.createFromBackendDict(
-                stateName, stateDict);
-              state.content.setHtml(
-                activeThread.getReplacementHtmlFromSuggestion());
-              if (result.audioUpdateRequired) {
-                state.contentIdsToAudioTranslations.markAllAudioAsNeedingUpdate(
-                  state.content.getContentId());
+        if (result.action === 'edit' &&
+                result.suggestionType === 'edit_exploration_state_content') {
+          url = UrlInterpolationService.interpolateUrl(
+            EDIT_SUGGESTION_URL_TEMPLATE, {
+              suggestion_id: activeThread.suggestion.suggestionId,
+            }
+          ),
+          data = {
+            action: result.action,
+            summary_message: result.summaryMessage,
+            change: {
+              cmd: 'edit_state_property',
+              property_name: 'content',
+              state_name: stateName,
+              old_value: result.oldContent,
```

Comment:
@import-keshav, I don't see any comment on this, can you please reply?

Reply:
Actually, I don't think it makes sense to send the old value back to the backend. It will be overwritten anyway when the suggestion is retrieved. It should not be stored. Only new_value is important.

Does it raise any errors if old_value isn't provided? @import-keshav where did you see the old_value pattern existing?

Reply:
@import-keshav, can you please reply on this?